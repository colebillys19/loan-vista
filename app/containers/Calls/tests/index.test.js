import React from 'react';
import { render } from 'react-testing-library';
import { ConnectedRouter } from 'connected-react-router';
import { Provider } from 'react-redux';

import configureStore from 'configureStore';
import history from 'utils/history';
import { Calls } from '../index';

// Create redux store with history
const initialState = {};
const store = configureStore(initialState, history);

const WrappedCalls = (props) => (
  <Provider store={store}>
    <ConnectedRouter history={history}>
      <Calls {...props} />
    </ConnectedRouter>
  </Provider>
);

describe('<Calls />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const dispatch = jest.fn();
    render(<WrappedCalls dispatch={dispatch} />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<WrappedCalls />);
    expect(firstChild).toMatchSnapshot();
  });
});
