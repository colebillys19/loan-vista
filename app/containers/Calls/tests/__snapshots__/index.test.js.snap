// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calls /> Should render and match the snapshot 1`] = `
<section
  aria-labelledby="calls-tab"
  class="TabWrapper-j2wsfb-0 beIMbY"
  id="calls-view"
>
  <div
    class="styledComponents__CallsViewWrapper-sc-122lc2r-0 iZhhoy"
  >
    CallsView
  </div>
</section>
`;
